classes/fetch_services.py

```diff
@@ -136,60 +148,78 @@ def request(self, url: str):
         # but according to its source code, it looks like it uses UTC:
         # <https://github.com/yt-dlp/yt-dlp/blob/07f5b2f7570fd9ac85aed17f4c0118f6eac77beb/yt_dlp/YoutubeDL.py#L2631>
         date_format = "%Y%m%d"
-        upload_date = datetime.strptime(response.get("upload_date"), date_format)
+        upload_date = datetime.strptime(video_data.get("upload_date"), date_format)
         upload_date = upload_date.replace(tzinfo=timezone.utc)
 
-        if response.get("duration") is None and self.prompt_dur:
-            mins = None
-            seconds = None
-            err("Missing Duration, Please Add Manually:")
-
-            while True:
-                try:
-                    if mins is None:
-                        mins = int(input("Minutes: "))
-                    seconds = int(input("Seconds: "))
-                    break
-                except:
-                    continue
-
-            response["duration"] = mins * 60 + seconds
-
-        if not response.get("idealized"):
-            # some sites will return a display name rather than unique
-            # usernames or they might return extra things in their
-            # response properties, so this is here to make adjustments
-            # for an ideal response, which will skip this step if loaded from the cache
-            match response["webpage_url_domain"]:
-                case "twitter.com":
-                    response["channel"] = response.get("uploader_id")
-                    response["title"] = response.get("title")[
-                        response.get("title").find("-") + 2 :
-                    ]
-                case "tiktok.com":
-                    response["channel"] = response.get("uploader")
-                    # could be attached to the if condition, but would feel
-                    # unorganizd if future sites are added that might need to use this
-
-            # bilibili.com and newgrounds.com sites don't seem to return information for the `channel`
-            # response property, so use `uploader` instead.
-            if response.get("channel") is None:
-                response["channel"] = response.get("uploader")
-
-            response["idealized"] = True
-
-        video_data = {
-            "title": response.get("title"),
-            "uploader": response.get("channel"),
+        return {
+            "title": video_data.get("title"),
+            "uploader": video_data.get("uploader"),
             "upload_date": upload_date,
-            "duration": response.get("duration"),
+            "duration": video_data.get("duration"),
         }
 
-        return video_data
+    # Some urls might have specific issues that should
+    # be handled here before they can be properly processed
+    # If yt-dlp gets any updates that resolve any of these issues
+    # then the respective case should be updated accordingly
+    def preprocess(self, url: str) -> dict | None:
```

Comment:
This method can't return `None`. It can return an empty dictionary, which is not the same as `None`.

Reply:
I think I've resolved everything mentioned here in the latest commit. Thanks for pointing all this out!